probes/hasGoldBadge/impl.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package hasGoldBadge
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const Probe = "hasGoldBadge"
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	r := raw.CIIBestPracticesResults
+
+	if r.Badge == clients.Gold {
+		return positiveOutcome()
+	}
+	return negativeOutcome()
+}
+
+func negativeOutcome() ([]finding.Finding, string, error) {
+	f, err := finding.NewWith(fs, Probe,
+		"Did not detect Gold badge.", nil,
+		finding.OutcomeNegative)
+	if err != nil {
+		return nil, Probe, fmt.Errorf("create finding: %w", err)
+	}
+	return []finding.Finding{*f}, Probe, nil
+}
+
+func positiveOutcome() ([]finding.Finding, string, error) {
+	f, err := finding.NewWith(fs, Probe,
+		"Detected Gold badge.", nil,
```

Comment:
`OpenSSF best practice badge found at Gold level`. Can change this wording in all probes

Reply:
Changed for all except Unknown badge.